### DISCUSSION TITLE
Global Throttle Limit - Attempt to make EU/UK legal throttle

### DISCUSSION THREAD

**Discussion by DavesDevFails** [danielnilsson9/bbs-fw#187] [Q&A]
I saw this on another issue/discussion but can't find it now.. 

I've just started to use the 'Global Throttle Limit' to try and create a UK legal throttle (can only provide power until 6 km/h when not pedalling, and up to 25 km/h when pedalling) on my BBSHD.

I saw @danielnilsson9 mention in the other thread that it currently wouldn't be able to work like that, because something always overrides the other? (Sorry trying to go from memory, I searched high and low and couldn't find the post again).

Looking at the **apply_speed_limit** function, I can't see why this hack wouldn't work 
(spoiler: I tested it, it doesn't work - I get the global speed limit whether pedalling or not..)

changing this section: 
```
	if (global_throttle_limit_active)
	{
		// use configured global throttle override speed limit
		max_speed_rpm_x10 = global_throttle_speed_limit_rpm_x10;
	}
```

to this:
```
	if (global_throttle_limit_active)
	{
		if (pas_is_pedaling_forwards())
		{
		// use configured global throttle override speed limit (25 km/h)
			max_speed_rpm_x10 = global_throttle_speed_limit_rpm_x10;
			
		}
		else 
		{
		// use uk/eu legal throttle only speed (6 km/h - hardcoded here, but could use assist level max speed or whatever) 
			max_speed_rpm_x10 = convert_wheel_speed_kph_to_rpm(6) * 10;
		}
	}
```

Anybody got any ideas?

**Comment by phil-83** [comment]
I'm not 100% sure, but looking at how pas_is_pedalling_forwards is set, it might stay true until the pedalling direction is changed. Maybe try getting some speed up, pedal a rev backwards then try the throttle to see if it works as desired.

I did post something similar to what you've done here on the endless sphere thread recently. Shame it wasn't as easy as I'd hoped.

**Reply by DavesDevFails** [reply]
I don't think that is the case, it's how the regular pedal assist feature works, it also checks the condition of _'pas_is_pedaling_forwards()'_ (among other things) before it sends power to the motor for pedal assist, it stops as soon as you stop pedalling (no backpedalling required), and restarts when you start pedalling again, so the 'pas_is_pedalling_forwards()' boolean seems to be working fine.

It has me very confused, but to be fair, most other things I've thought _'that should be easy to implement'_ and then tried to do it myself haven't worked (well) either.

**Comment by danielnilsson9** [comment]
The discussion was in issue #112

But yes, there are some conflicting views on this, I am considering making the behaviour configurable.

My country (Sweden) does not allow throttle at all as far as I know but walk mode is allowed. So I made the behaviour pretty much the same as walk mode (PAS disabled while active).

But for a smooth throttle start in PAS, the other way around would definitely be better which is probably what you want...
